server/tests/order/test_service.py

```diff
@@ -1818,3 +1819,91 @@ async def test_process_dunning_order_success(
             order_id=order.id,
             payment_method_id=payment_method.id,
         )
+
+
+@pytest.mark.asyncio
+class TestTriggerPayment:
+    """Test payment lock mechanism in trigger_payment service method."""
+
+    async def test_trigger_payment_when_already_locked(
+        self,
+        session: AsyncSession,
+        save_fixture: SaveFixture,
+        product: Product,
+        customer: Customer,
+    ) -> None:
+        """Test that trigger_payment raises PaymentAlreadyInProgress when order is already locked."""
+        # Given
+        payment_method = await create_payment_method(save_fixture, customer=customer)
+        order = await create_order(
+            save_fixture,
+            product=product,
+            customer=customer,
+            status=OrderStatus.pending,
+        )
+        order.payment_lock_acquired_at = utc_now()
+        await save_fixture(order)
+
+        # When/Then
+        with pytest.raises(PaymentAlreadyInProgress):
+            await order_service.trigger_payment(session, order, payment_method)
+
+    async def test_trigger_payment_acquires_lock_successfully(
+        self,
+        stripe_service_mock: MagicMock,
+        session: AsyncSession,
+        save_fixture: SaveFixture,
+        product: Product,
+        customer: Customer,
+    ) -> None:
+        """Test that trigger_payment acquires lock and processes payment normally."""
+        # Given
+        payment_method = await create_payment_method(save_fixture, customer=customer)
+        order = await create_order(
+            save_fixture,
+            product=product,
+            customer=customer,
+            status=OrderStatus.pending,
+        )
+        order.customer.stripe_customer_id = "cus_test123"
```

Comment:
`customer` fixture should already have a Stripe customer ID.

Reply:
Foxed